packages/Modal/src/index.tsx

```diff
@@ -20,64 +26,81 @@ const Modal: React.SFC<IModalProps> = ({
   actions,
   isActive,
   buttonText,
-  buttonModifiers
-}) => {
-  const [modalStatus, setModalStatus] = useState(isActive);
-  const type = () => {
-    if (simple) {
-      return "--simple";
-    }
-    if (alert) {
-      return "--alert";
-    }
-    if (upload) {
-      return "--upload";
-    }
-    return "";
+  buttonModifiers,
+  utils: { classNames },
+  onClose = () => ({})
+}): JSX.Element => {
+  const [modalStatus, setModalStatus] = useState<boolean | undefined>(isActive);
+
+  const closeModal = (): void => {
+    setModalStatus(false);
+    onClose();
   };
-  const renderModal = (
+
+  const modalClassNames: string = classNames("modal modal--show", {
+    "modal--alert": alert,
+    "modal--simple": simple,
+    "modal--upload": upload
+  });
+
+  const [
+    modalHeaderClassNames,
+    modalContentClassNames,
+    modalFooterClassNames
+  ]: string[] = ["header", "content", "footer"].map(
+    (alertType: string): string =>
+      classNames(`modal__${alertType}`, {
+        [`modal__${alertType}--alert`]: alert,
+        [`modal__${alertType}--simple`]: simple,
+        [`modal__${alertType}--upload`]: upload
+      })
+  );
+
+  return (
     <Fragment>
-      <div className="overlay" />
-      <div className={`modal modal${type()} modal--show`}>
-        <div className={`modal__header modal__header${type()}`}>
-          {!alert && <div className="modal__title">{title}</div>}
-          {!alert && (
-            <div
-              className="modal__close"
-              role="button"
-              onClick={() => setModalStatus(false)}
-            >
-              <i className="material-icons">close</i>
+      {modalStatus && (
+        <div className="overlay">
+          <div className={modalClassNames}>
+            <div className={modalHeaderClassNames}>
+              {!alert && <div className="modal__title">{title}</div>}
+              {!alert && (
+                <div
```

Comment:
You could do just one alert condition instead two, and wrap the content inside a fragment